### PR TITLE
add setup based db tests

### PR DESCRIPTION
**Pull request by PrateekGarg-gs**
- object naming tests
- group By tests
- slice, drop, limit, top functions

### CODE REVIEW

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/tests/dbSpecificTests/h2Tests.pure [hunk 1]
```diff
@@ -29,12 +29,51 @@ function <<test.TestCollection>> meta::relational::tests::dbSpecificTests::H2::r
...
 {
   //optionally assert db specific sql for a bunch of tests
    let expectedSqls= [
      pair(meta::relational::tests::dbSpecificTests::sqlQueryTests::dynaFunctions::abs::testNegative_DbTestConfig_1__Boolean_1_, 'select abs(-1)')
    ]->newMap();
 
   ^$dbTestConfig(expectedSql= $expectedSqls->get($f));
+}
+
+function <<test.TestCollection>> meta::relational::tests::dbSpecificTests::H2::aggregateFunctions::runAggregateFunctionTests(): PureTestCollection[1]
```
Comment (PrateekGarg-gs) [inline legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/tests/dbSpecificTests/h2Tests.pure:42]: use one test collection

File: legend-engine-server/src/test/java/org/finos/legend/engine/server/test/pureClient/stores/dbSpecific/Relational_DbSpecific_UsingPureClientTestSuite.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ public static Test createSuite(String pureTestCollectionPath, String testServerC
         final ThreadLocal<ServersState> state = new ThreadLocal<>();
         state.set(PureTestHelper.initEnvironment(false, testServerConfigFilePath));
 
-        CompiledExecutionSupport executionSupport = getClassLoaderExecutionSupport();
+        CompiledExecutionSupport executionSupport = getClassLoaderExecutionSupport(true);
```
Comment (vikask-gs) [inline legend-engine-server/src/test/java/org/finos/legend/engine/server/test/pureClient/stores/dbSpecific/Relational_DbSpecific_UsingPureClientTestSuite.java:42]: why this change?

File: legend-engine-server/src/test/java/org/finos/legend/engine/server/test/shared/PureTestHelper.java [hunk 3]
```diff
@@ -141,6 +141,12 @@ public static ServersState initEnvironment(boolean withH2, String serverConfigFi
             h2Server = AlloyH2Server.startServer(relationalDBPort);
             System.out.println("H2 database started on port:" + relationalDBPort);
         }
+        else
+        {
```
Comment (vikask-gs) [inline legend-engine-server/src/test/java/org/finos/legend/engine/server/test/shared/PureTestHelper.java:145]: why this change?

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/dbTestRunner/shared.pure [hunk 4]
```diff
@@ -54,19 +55,57 @@ function meta::relational::dbTestRunner::collectTests(pack: Package[1]): Functio
...
+       let basicPlan = if($m.name->isEmpty(),
+                      |meta::pure::executionPlan::executionPlan($f, ^ExecutionContext(), $extensions, noDebug());,
+                      |meta::pure::executionPlan::executionPlan($f, $m, $pureRuntime, ^ExecutionContext(),$extensions, noDebug()););
+       []; ,               
+     | 
+       let conn = $config.connection->cast(@RelationalDatabaseConnection)->toOne();
+       let connWithDb= ^$conn(element=$db);
+       let pureRuntime = ^Runtime(connections=$connWithDb);
+       
+       execute($f, $m, $pureRuntime, ^ExecutionContext(), $extensions);
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/dbTestRunner/shared.pure:109]: I thought we discussed that it should be a 2 step process, where you will first calculate the plan and then execute it?

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/dbTestRunner/shared.pure [hunk 5]
```diff
@@ -54,19 +55,57 @@ function meta::relational::dbTestRunner::collectTests(pack: Package[1]): Functio
...
+}
+
+function meta::relational::dbTestRunner::executeViaPlan<T|m>(f:FunctionDefinition<{->T[m]}>[1], m:Mapping[1], db:meta::relational::metamodel::Database[1], config:DbTestConfig[1], extensions:meta::pure::extension::Extension[*]):meta::pure::mapping::Result<T|m>[0..1]
+{
+  if($config.connection->isEmpty(),
+     | 
+       println('**** Warning **** : No connection available for given dbtype, Skipping execution of plan');
+       let pureRuntime = ^Runtime(connections= ^TestDatabaseConnection(element = $db, type= $config.dbType));
+
+       let basicPlan = if($m.name->isEmpty(),
```
Comment (vikask-gs) [inline legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/dbTestRunner/shared.pure:100]: what does this mean?

File: legend-engine-server/src/test/java/org/finos/legend/engine/server/test/shared/PureTestHelper.java (hunk 3 above)
Comment (PrateekGarg-gs) [reply legend-engine-server/src/test/java/org/finos/legend/engine/server/test/shared/PureTestHelper.java:145]: so that we only set planLocal and exceplan properties for db tests , shouldnt affect other tests

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/dbTestRunner/shared.pure (hunk 4 above)
Comment (PrateekGarg-gs) [reply legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/dbTestRunner/shared.pure:109]: this actually does that via legendExecute function, Earlier I tried at this place, there were issues with type casting Result<Any|*> cant be cast to Result<T|m> . ( executePlan returns Result<ANy|*>. Copying results to a new object was very slow also)

File: legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/dbTestRunner/shared.pure (hunk 5 above)
Comment (PrateekGarg-gs) [reply legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/dbTestRunner/shared.pure:100]: it can be a case where we dont supply any mapping/runtime. ( going to be supported in engine as far as I understand)

File: legend-engine-server/src/test/java/org/finos/legend/engine/server/test/pureClient/stores/dbSpecific/Relational_DbSpecific_UsingPureClientTestSuite.java (hunk 2 above)
Comment (PrateekGarg-gs) [reply legend-engine-server/src/test/java/org/finos/legend/engine/server/test/pureClient/stores/dbSpecific/Relational_DbSpecific_UsingPureClientTestSuite.java:42]: enables console for tests in compiled mode. just to print out some stms like 'server not available' or what error occurred. Rather than just test failed/ assert failed.

**Comment by vikask-gs** [comment]
Good to merge. NOTE: Pending Tc run

**Review by gs-ssh16 — APPROVED** [review]
